grafana/consul-k8s-control-plane-monitoring.json

```diff
@@ -1744,6 +1744,180 @@
                     ],
                     "title": "Received bytes total per 5 minutes (Consul servers)",
                     "type": "timeseries"
+                },
+                {
+                    "datasource": {
+                        "type": "prometheus",
+                        "uid": "PBFA97CFB590B2093"
+                    },
+                    "description": "Gives the filesystem storage used by the container",
+                    "fieldConfig": {
+                        "defaults": {
+                            "color": {
+                                "mode": "continuous-BlYlRd"
+                            },
+                            "mappings": [],
+                            "thresholds": {
+                                "mode": "absolute",
+                                "steps": [
+                                    {
+                                        "color": "green",
+                                        "value": null
+                                    },
+                                    {
+                                        "color": "red",
+                                        "value": 80
+                                    }
+                                ]
+                            },
+                            "unit": "bytes"
+                        },
+                        "overrides": []
+                    },
+                    "gridPos": {
+                        "h": 8,
+                        "w": 9,
+                        "x": 15,
+                        "y": 36
+                    },
+                    "id": 100,
+                    "options": {
+                        "orientation": "auto",
+                        "reduceOptions": {
+                            "calcs": [
+                                "lastNotNull"
+                            ],
+                            "fields": "",
+                            "values": false
+                        },
+                        "showThresholdLabels": false,
+                        "showThresholdMarkers": true
+                    },
+                    "pluginVersion": "9.5.5",
+                    "targets": [
+                        {
+                            "datasource": {
+                                "type": "prometheus",
+                                "uid": "PBFA97CFB590B2093"
+                            },
+                            "editorMode": "code",
+                            "exemplar": false,
+                            "expr": "sum(container_fs_usage_bytes)",
```

Comment:
I think this should be `sum(container_fs_usage_bytes{consul_agent_type="server"}) by (instance)` since we only concern about the instance running consul servers

Reply:
I do not see `consul_agent_type` in container_fs_usage_bytes

In local setup, I see below for the prometheus query container_fs_usage_bytes
```container_fs_usage_bytes{beta_kubernetes_io_arch="arm64", beta_kubernetes_io_os="linux", device="/run/containerd/io.containerd.grpc.v1.cri/sandboxes/9c7e2384d3fbbed1d9a7e3d91dc7652ce68dbc81b82169036fee5b16e8080a87/shm", id="/", instance="kind-control-plane", job="kubernetes-nodes-cadvisor", kubernetes_io_arch="arm64", kubernetes_io_hostname="kind-control-plane", kubernetes_io_os="linux"}```

In AWS setup, I see below for the prometheus query container_fs_usage_bytes
```container_fs_usage_bytes{app="load _generator", beta_kubernetes_io_arch-"amd64", beta_kubernetes_io_instance_type="3a.medium", beta_kubernetes_io_os="linux", device="/dev/nvmeOn1p1", eks_amazonaws_com_capacityType="ON_DEMAND", eks_amazonaws_com_nodegroup="load_generator-yuix-20240111084441808900000027",eks_amazonaws_com_nodegroup_image="ami-012689cd52612e266",eks_amazonaws_com_sourceLaunchTemplateld="It-038335641480e3bd0", eks_amazonaws_com_sourceLaunchTemplateVersion="",failure_domain_beta_kubernetes_io_region-"us-east-2",failure_domain_beta_kubernetes_io_zone="us-east-2c",id="/",instance="p-10-0-6-141.us-east-2.compute.internal",job="kubernetes-nodes-cadvisor", k8s_io_cloud_provider_aws="dde6abeafc395c33548327bdeb9d3c97", kubernetes_io_arch-"amd64", kubernetes_io_hostname="ip-10-0-6-141.us-east-2.compute.internal", kubernetes_jo_os="linux' node_kubernetes_io_instance_type="t3a.medium", topology_ebs_csi_aws_com_zone="us-east-2c", topology_kubernetes_io_region="us-east-2", topology_kubernetes_io_zone="us-east-2c"}```

Reply:
I see. `consul_agent_type` is a tag we added to mark the k8s node to run consul servers.  I don't think we can add `by instance`. It will cause multiple metrics in the gauge.
![Screenshot 2024-01-12 at 2 09 15 PM](https://github.com/hashicorp/consul/assets/463631/7dbb6880-17a1-43c3-bd31-92f2d34ef50f)

So, I think the usage should also be a line chart rather than a gauge. Let's use `sum(container_fs_usage_bytes) by (instance)` in a line graph

Reply:
I created a new AWS setup and see consul_agent_type tag. Not really sure why it was not shown earlier :shrug: . We have two options
1. sum(container_fs_usage_bytes{consul_agent_type="server"}) by (instance) as line graph which gives disk usage per instance as linegraph
<img width="1708" alt="Screenshot 2024-01-15 at 11 43 28 AM" src="https://github.com/hashicorp/consul/assets/3773964/c78c846a-e2ea-4c60-9592-2dd31b903ef3">

2. sum(container_fs_usage_bytes{consul_agent_type="server"}) by (consul_agent_type) which gives disk usage of all consul servers put together as a guage
<img width="1722" alt="Screenshot 2024-01-15 at 11 34 51 AM" src="https://github.com/hashicorp/consul/assets/3773964/ceb78ff1-a38a-4d70-bca1-fd33d0013d69">

I believe going ahead with guage for all consul servers gives the required information. Let me know if you have different thoughts